### PR TITLE
Add terratest plumbing

### PR DESCRIPTION
**Pull request by jsclarridge**
Per https://www.pivotaltracker.com/story/show/169407534, introduce Terratest into the module (the module was already upgraded to TF 12 syntax). The PR also includes:

Makefile - For running pre-commit and Terratest
golangci-lint - Go linting
dependabot - To keep go modules up to date
README - Updates for how to run tests

The Terratest example lives in examples/simple. Running against our CI AWS account returns

--- PASS: TestTerraformAwsNlb (498.50s)
PASS
ok  	github.com/trussworks/terraform-aws-nlb-containers/test	498.517s

### CODE REVIEW

File: examples/simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+module "vpc" {
+  source             = "terraform-aws-modules/vpc/aws"
+  version            = "~> 2"
+  cidr               = "10.0.0.0/16"
+  azs                = var.vpc_azs
+  private_subnets    = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets     = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+  enable_nat_gateway = true
```
Comment (dynamike) [inline examples/simple/main.tf]: I don't think you need a NAT Gateway for this test. 
```suggestion

```

File: examples/simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  region          = var.region
+  nlb_logs_prefix = "nlb/${var.nlb_name}-${var.environment}"
+}
+
+module "vpc" {
+  source             = "terraform-aws-modules/vpc/aws"
+  version            = "~> 2"
+  cidr               = "10.0.0.0/16"
+  azs                = var.vpc_azs
+  private_subnets    = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
```
Comment (dynamike) [inline examples/simple/main.tf]: I don't think you'll need private subnets for this test either. 
```suggestion

```

File: examples/simple/main.tf [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  nlb_subnet_ids = module.vpc.public_subnets
+  nlb_vpc_id     = module.vpc.vpc_id
+}
+
+module "nlb_logs" {
+  source          = "trussworks/logs/aws"
+  version         = "~> 4"
+  s3_bucket_name  = var.logs_bucket
+  region          = var.region
+  nlb_logs_prefix = "nlb/${var.nlb_name}-${var.environment}"
```
Comment (dynamike) [inline examples/simple/main.tf]: blerg. I'll make a story to make this more flexible.

File: examples/simple/variables.tf [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  type = string
+}
+
+variable "region" {
+  type = string
+}
+
+variable "vpc_azs" {
+  type = list(string)
+}
```
Comment (kilbergr) [inline examples/simple/variables.tf]:
```suggestion
}

```

File: examples/simple/main.tf (hunk 1 above)
Comment (jsclarridge) [reply examples/simple/main.tf]: Removed.

File: examples/simple/main.tf (hunk 2 above)
Comment (jsclarridge) [reply examples/simple/main.tf]: Removed.

File: examples/simple/variables.tf (hunk 4 above)
Comment (jsclarridge) [reply examples/simple/variables.tf]: Done.

**Pull request merged**